pkg/odo/genericclioptions/context.go

```diff
@@ -147,6 +186,46 @@ func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
 			}
 		}
 	}
+	return app
+}
+
+// resolveComponent resolves component
+func resolveComponent(command *cobra.Command, configFileName string, context *Context) string {
+	var cmp string
+	lci, err := config.NewLocalConfigInfo(configFileName, false)
+	util.LogErrorAndExit(err, "could not get component settings from config file")
+	cmpFlag := FlagValueIfSet(command, ComponentFlagName)
+	if len(cmpFlag) == 0 {
+		// retrieve the current component if it exists if we didn't set the component flag
+		cmp = lci.GetName()
+	} else {
+		// if flag is set, check that the specified component exists
+		context.checkComponentExistsOrFail(cmpFlag)
+		cmp = cmpFlag
+	}
+	return cmp
+}
+
+// newContext creates a new context based on the command flags, creating missing app when requested
+func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
+	client := client(command)
+
+	// resolve project
+	ns := resolveProject(command, client)
```

Comment:
Can this be done below after the config resolution?

Reply:
done